src/writer.rs

```diff
@@ -0,0 +1,168 @@
+//! Writer to serialize an [`Element`] (its name, attributes and children).
+//!
+//! [`Element`]: ../minidom/element/struct.Element.html
+
+use crate::Error;
+use minidom::{Element, Node};
+use quick_xml::{
+    events::{BytesDecl, BytesEnd, BytesStart, BytesText, Event},
+    Writer,
+};
+use std::io::Write;
+const XML_VERSION: &str = "1.0";
+const ENCODING: &str = "UTF-8";
+
+/// Writer for [`Element`].
+///
+/// [`Element`]: ../minidom/element/struct.Element.html
+pub struct ElementWriter<W>
+where
+    W: Write,
+{
+    writer: Writer<W>,
+}
+
+// Publicly exposed methods
+impl<W> ElementWriter<W>
+where
+    W: Write,
+{
+    /// Creates a [`ElementWriter`] from an instance of [`Write`].
+    ///
+    /// This writer writes the [`Element`] with no spaces or new lines.
+    ///
+    /// ```plain
+    /// <?xml version="1.0" encoding="UTF-8"?><todos><todo>Write documentation</todo><todo>Release 'minidom_writer'</todo></todos>
+    /// ```
+    ///
+    /// [`Element`]: ../minidom/element/struct.Element.html
+    /// [`ElementWriter`]: struct.ElementWriter.html
+    /// [`Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
+    pub fn plain(write: W) -> Self {
+        ElementWriter {
+            writer: Writer::new(write),
+        }
+    }
+
+    /// Creates a [`ElementWriter`] from an instance of [`Write`].
+    ///
+    /// This writer writes the [`Element`] with indentation.
+    ///
+    /// ```plain
+    /// <?xml version="1.0" encoding="UTF-8"?>
+    /// <todos>
+    ///     <todo>Write documentation</todo>
+    ///     <todo>Release 'minidom_writer'</todo>
+    /// </todos>
+    /// ```
+    ///
+    ///
+    /// [`Element`]: ../minidom/element/struct.Element.html
+    /// [`ElementWriter`]: struct.ElementWriter.html
+    /// [`Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
+    pub fn pretty(write: W) -> Self {
+        ElementWriter {
+            // 9 is ASCII code for Tabulation
+            writer: Writer::new_with_indent(write, 9, 1),
+        }
+    }
+
+    /// Consumes this [`ElementWriter`], returning the underlying [`Write`].
+    ///
+    /// [`ElementWriter`]: struct.ElementWriter.html
+    /// [`Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
+    pub fn into_inner(self) -> W {
+        self.writer.into_inner()
+    }
+
+    /// Write the [`Element`] into a implementation of [`Write`].
```

Comment:
```suggestion
    /// Writes the [`Element`] into an implementation of [`Write`].
```

Reply:
Done in 100a6327091ff6989067030cd85ae9ae66a20c79. Thanks.